### PR TITLE
[Update] Adding a method to create an entity naturally

### PR DESCRIPTION
**Pull request by rexbut**
[API](https://github.com/SpongePowered/SpongeAPI/pull/1350) | Common

To fix conflicts, I recreated a new branch.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java [hunk 1]
```diff
@@ -463,6 +463,12 @@ public Entity createEntity(EntityType type, Vector3d position) throws IllegalArg
         checkRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
+    
+    @Override
+    public Entity createEntityNaturally(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkRange(position.getX(), position.getY(), position.getZ());
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java]: Insert `checkNotNull(position, "position");` before this line.

File: src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java [hunk 2]
```diff
@@ -499,6 +499,12 @@ public Entity createEntity(EntityType type, Vector3d position) throws IllegalArg
         checkRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
+    
+    @Override
+    public Entity createEntityNaturally(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkRange(position.getX(), position.getY(), position.getZ());
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java]: Insert `checkNotNull(position, "position");` before this line.

Comment (rexbut) [reply src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java]: I 've added other

File: src/main/java/org/spongepowered/common/world/extent/ExtentViewTransform.java [hunk 3]
```diff
@@ -479,8 +480,15 @@ private Vector3d inverseTransform(Vector3d vector) {
 
     @Override
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         return this.extent.createEntity(type, inverseTransform(position));
     }
+
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/common/world/extent/ExtentViewTransform.java]: Remove trailing whitespace

File: src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java [hunk 4]
```diff
@@ -496,9 +497,17 @@ public boolean spawnEntity(Entity entity, Cause cause) {
 
     @Override
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         checkRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
+
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java]: Remove trailing whitespace

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 5]
```diff
@@ -427,6 +437,11 @@ public Entity createEntity(EntityType type, Vector3d position) throws IllegalArg
                 return Optional.empty();
             }
         }*/
+
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: Remove trailing whitespace

File: src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java [hunk 6]
```diff
@@ -460,9 +461,17 @@ public boolean spawnEntity(Entity entity, Cause cause) {
 
     @Override
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         checkRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
+
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java]: Remove trailing whitespace

**Comment by gabizou** [comment]
Condense the three commits into 1 please.

**Comment by rexbut** [comment]
I just made a rebase.

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 7]
```diff
@@ -433,6 +443,11 @@ public Entity createEntity(EntityType type, Vector3d position) throws IllegalArg
             }
         }*/
 
+        if (naturally && entity instanceof EntityLiving) {
+            // Adding the default equipment
+            ((EntityLiving)entity).onInitialSpawn(world.getDifficultyForLocation(new BlockPos((net.minecraft.entity.Entity) entity)), null);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: Need some proper spacing. Also, why are you using the entity constructor when you already have the `x`, `y`, and `z`?

File: src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java [hunk 8]
```diff
@@ -458,11 +459,19 @@ public boolean spawnEntity(Entity entity, Cause cause) {
...
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         checkBlockRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
 
     @Override
+    public Entity createEntityNaturally(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
+        checkBlockRange(position.getX(), position.getY(), position.getZ());
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java]: No.... You don't want to check block range..... Entities can exist outside the bounds of blocks.

File: src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java [hunk 9]
```diff
@@ -494,11 +495,19 @@ public boolean spawnEntity(Entity entity, Cause cause) {
...
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         checkBlockRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
 
     @Override
+    public Entity createEntityNaturally(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
+        checkBlockRange(position.getX(), position.getY(), position.getZ());
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/world/extent/SoftBufferExtentViewDownsize.java]: And here, don't check block range.

File: src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java (hunk 8 above)
Comment (kashike) [reply src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java]: This is what `createEntity` is doing, so those methods should change too in this case.

**Pull request merged**